File: backend/IMIS/IMIS.Domain/PgsSignatoryTemplate .cs
using Base.Primitives;

namespace IMIS.Domain
{
    public class PgsSignatoryTemplate : Entity<int>
    {
        public required string Status { get; set; }
        public required string SignatoryLabel { get; set; }
        public int OrderLevel { get; set; }
        public string? DefaultSignatoryId { get; set; }
        public User? DefaultSignatory { get; set; }
        public bool IsActive { get; set; }
    }
}

File: frontend/imis/lib/performance_governance_system/pgs_signatory_template/models/pgs_signatory_template.dart
import 'package:json_annotation/json_annotation.dart';
part 'pgs_signatory_template.g.dart';

@JsonSerializable()
class PgsSignatoryTemplate {
  int? id;
  bool isDeleted;
  String? status;
  int orderLevel;
  String defaultSignatoryId;
  bool isActive;
  String? rowVersion;
  String? signatoryLabel;

  PgsSignatoryTemplate(
    this.id,
    this.isDeleted,
    this.orderLevel,
    this.defaultSignatoryId,
    this.isActive, {
    required this.status,
    this.rowVersion,
    required this.signatoryLabel,
  });

  factory PgsSignatoryTemplate.fromJson(Map<String, dynamic> json) =>
      _$PgsSignatoryTemplateFromJson(json);

  Map<String, dynamic> toJson() => _$PgsSignatoryTemplateToJson(this);
}
